lib/compiler.js

```diff
@@ -598,3 +604,69 @@ function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
   var fixedFile = fixFileExtension(resolvedPath, files, false);
   return (fixedFile === undefined ? resolvedPath : fixedFile);
 }
+
+function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
+  var extensions = _.without(_.keys(require.extensions),
+    _.keys(getExcludedExtensions()));
+  var files = options.mixins || [];
+
+  mixinDirs.forEach(function(dir) {
+    dir = tryResolveAppPath(appRootDir, dir);
```

Comment:
Per the comment in the code 
```
   * `resolveOptions.strict` = true,
   * means retain backward compatibility when resolving the path and
   * `resolveOptions.strict` = false,
   * does not enforce any such restriction when resolving the path
```
When you say
> explicitly set strict:false only in the few places where we need to preserve backwards compatibility.

I'm a bit confused, as ```strict:true``` retains backward compatibility.